source/m2mbase.cpp

```diff
@@ -91,18 +91,15 @@ M2MBase::M2MBase(const String& resource_name,
             }
         }
 
-        _sn_resource->name = stringdup((char*)resource_name.c_str());
+        if((!resource_name.empty())) {            
+            _sn_resource->identifier_int_type = false;
+            _sn_resource->identifier.name = stringdup((char*)resource_name.c_str());            
+        } else {
+            tr_debug("M2MBase::M2Mbase resource name is EMPTY ===========");
+            _sn_resource->identifier_int_type = true;
+        }        
```

Comment:
Is identifier.instance_id left uninitialized on else-branch on purpose? On the other hand, what is the default value if one forgets to set the id later? 0?